testkit/shared/src/main/scala/org/typelevel/otel4s/testkit/MetricData.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.testkit
+
+sealed trait MetricData extends Product with Serializable
+
+object MetricData {
+  final case class QuantileData(quantile: Double, value: Double)
+
+  final case class SummaryPoint(
+      sum: Double,
+      count: Long,
+      values: List[QuantileData]
+  )
+
+  final case class HistogramPoint(
+      sum: Double,
+      count: Long,
+      boundaries: List[Double],
+      counts: List[Long]
+  )
+
+  final case class LongGauge(points: List[Long]) extends MetricData
```

Comment:
It's a bit of a lie, actually. 

The real data is:
```scala
case class LongGauge(points: List[LongGauge.PointData]) extends MetricData

object LongGauge {
  case class PointData(
    startEpochNanos: Long,
    epochNanos: Long,
    value: Long,
    attributes: List[Attribute[_]],
    exemplars: List[Exampelars]
  )
}
```

But I assume, in the majority of cases, you will need only `value` and `attributes`. 

As an option, there can be two models: 
- `Full` - 1 to 1 projection of Java Otel models. 
- `Minimal/Simple` - only necessary information (i.e. `value`, `attributes`) 

Then, the metric data can be described as:
```scala
case class LongGauge(points: List[LongGauge.PointData]) extends MetricData

object LongGauge {
  case class PointDataFull(
    startEpochNanos: Long,
    epochNanos: Long,
    value: Long,
    attributes: List[Attribute[_]],
    exemplars: List[Exampelars]
  ) {
    def toSimple: PointDataSimple = ...
  }
  
  case class PointDataSimple(value: Long, attributes: List[Attribute[_]])
}
```

Reply:
Seems a user can define a needed projection on a project level. Current models fit fine for testing purposes. 
I'm not even sure about switching to `final class` from `final case class`. I don't think open telemetry specification is going to be changed really often, isn't it?

Reply:
I don't think they'll add required fields, but it wouldn't shock me if they add optional fields.

Reply:
Is this the last open design decision?  Whether we want to commit to case classes and their bincompat problems in the testkit?  I can do the grunt work of making this a builder.  If we have one that's abstract for futureproofing, there's really nothing wrong with providing case classes as the "projections".

Reply:
See an alternative version https://github.com/iRevive/otel4s/pull/1, please.  
`final class` looks good to me.

Reply:
That looks good to me.  I guess the question is, if we added new fields, would we add them to the Hash.  But that can be a question for whenever that day comes.